charm/src/charm.py

```diff
@@ -90,7 +105,7 @@ def _on_mongodb_client_relation_changed(
         if initial != self._stored.reldata["mongodb"]:
             self._update_layer_and_restart(None)
 
-    def _update_layer_and_restart(self, event) -> None:
+    def _update_layer_and_restart(self, _) -> None:
```

Comment:
I’m wondering why you need to specify an argument you won't use in a private method. Can we just not pass it?

Reply:
Ok, done, and added a few comments around all of this _ event nonsense. Let me know what you think.